src/test/java/com/example/javaee_jpa_hibernate/repository/InvoiceRepositoryTest.java

```diff
@@ -0,0 +1,170 @@
+package com.example.javaee_jpa_hibernate.repository;
+
+import com.example.javaee_jpa_hibernate.model.Invoice;
+import com.example.javaee_jpa_hibernate.model.InvoiceBody;
+import com.example.javaee_jpa_hibernate.model.counterparty.Counterparty;
+import com.example.javaee_jpa_hibernate.model.invoice_item.InvoiceItem;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Persistence;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class InvoiceRepositoryTest {
+
+    @Test
+    public void shouldCreateAnInvoice() throws Exception {
+        //given
+        EntityManagerFactory entityManagerFactory = Persistence
+                .createEntityManagerFactory("Invoices");
+        InvoiceRepository repository = new InvoiceRepository(entityManagerFactory);
+        Invoice invoice = InvoiceGenerator.getAnInvoice(LocalDate.now(), "Dummy Company", "Dummy City", Arrays.asList("Item one", "Item two"));
+        repository.getEntityManager().getTransaction().begin();
```

Comment:
That's one of the reasons why the repository should get EntityManager from outside, not create it by itself. There is no need - more - we strongly avoid creating getters that are not critically needed. EntityManager is a repository dependency and (at least for now) there shouldn't be accessible from outside, not provided by the repository.

So - please create EntityManager in a test case, and provide it to the repository via its constructor. This way you can use it also in the test because you will have a handler to this object.

Reply:
I allowed myself  to polish your English a bit:
That's one of the reasons why the repository should get EntityManager from outside, rather than create it by itself. 
There is no need for it, what is more, we strongly avoid creating getters that are not critically needed. 
EntityManager is a repository dependency and (at least for now) it shouldn't be accessible from outside, but(???) provided by the repository.

Reply:
Thanks! :)